### ISSUE TITLE
mysql datetime format still wrong, version 0.2.5

### ISSUE THREAD

**Issue by RainerEmrich** [nextcloud/deck#341]
### Steps to reproduce
1. Insert due date for a card.
2. Close card.
3. Reopen card.

### Expected behaviour
The date and time should be saved.

### Actual behaviour
The date and time are not saved.

### Server configuration

**Operating system:** Ubuntu 16.04.

**Web server:** Apache 2.4 (fpm-fcgi)

**Database:** Mariadb 10.2.10

**PHP version:** 7.1.11-1+ubuntu16.04.1+deb.sury.org+1
<details>
        <summary>PHP-modules loaded</summary>

```
 - Core
 - date
 - libxml
 - openssl
 - pcre
 - zlib
 - filter
 - hash
 - Reflection
 - SPL
 - session
 - standard
 - cgi-fcgi
 - mysqlnd
 - PDO
 - xml
 - apcu
 - apc
 - bz2
 - calendar
 - ctype
 - curl
 - dom
 - mbstring
 - fileinfo
 - ftp
 - gd
 - gettext
 - gmp
 - iconv
 - igbinary
 - imagick
 - imap
 - intl
 - json
 - ldap
 - exif
 - mcrypt
 - msgpack
 - mysqli
 - pdo_mysql
 - pdo_pgsql
 - pdo_sqlite
 - pgsql
 - Phar
 - posix
 - readline
 - redis
 - shmop
 - SimpleXML
 - smbclient
 - sockets
 - sqlite3
 - sysvmsg
 - sysvsem
 - sysvshm
 - tokenizer
 - wddx
 - xmlreader
 - xmlwriter
 - xsl
 - zip
 - memcached
 - libsmbclient
 - Zend OPcache
```
</details>

**Nextcloud version:** 12.0.3 - 12.0.3.3

**Where did you install Nextcloud from:** Updated from 12.0.2 via web updater

**Signing status:** No errors have been found.

**List of activated apps:**
<details>
        <summary>Enabled apps</summary>

```
 - activity: 2.5.2
 - admin_audit: 1.2.0
 - admin_notifications: 1.0.0
 - announcementcenter: 3.1.0
 - apporder: 0.4.0
 - audioplayer: 2.1.0
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.6
 - checksum: 0.3.5
 - circles: 0.13.6
 - comments: 1.2.0
 - contacts: 2.0.1
 - dav: 1.3.0
 - deck: 0.2.5
 - external: 2.0.3
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_accesscontrol: 1.2.5
 - files_automatedtagging: 1.2.2
 - files_downloadactivity: 1.1.1
 - files_markdown: 2.0.1
 - files_pdfviewer: 1.1.1
 - files_reader: 1.0.4
 - files_retention: 1.1.2
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - gpxedit: 0.0.7
 - gpxpod: 2.2.0
 - groupfolders: 1.1.0
 - issuetemplate: 0.2.2
 - keeweb: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - mail: 0.7.4
 - metadata: 0.6.0
 - nextcloud_announcements: 1.1
 - notes: 2.3.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - previewgenerator: 1.0.7
 - provisioning_api: 1.2.0
 - quota_warning: 1.1.1
 - ransomware_protection: 1.0.5
 - richdocuments: 1.12.36
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - spreed: 2.0.1
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0
```
</details>

<details>
        <summary>Disabled apps</summary>

```
 - encryption
 - files_external
 - radio
 - user_external
 - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
        <summary>Content of config/config.php</summary>

```
{
    "instanceid": "ocldkw9697wg",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "testcloud.emrich-ebersheim.de"
    ],
    "datadirectory": "\/var\/ncdata\/nc_01\/data",
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "12.0.3.3",
    "dbname": "nc_cloud01",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "UTC",
    "installed": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "\/var\/run\/redis\/redis.sock",
        "port": 0,
        "timeout": 0,
        "dbindex": 0
    },
    "filelocking.enabled": true,
    "appstore.experimental.enabled": "true",
    "updater.server.url": "https:\/\/updates.nextcloud.org\/updater_server\/",
    "updater.release.channel": "stable",
    "mail_smtpmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "localhost",
    "mail_smtpport": "25",
    "enable_previews": true,
    "preview_max_x": 1024,
    "preview_max_y": 1024,
    "preview_max_scale_factor": 8,
    "trashbin_retention_obligation": "auto",
    "maintenance": false,
    "theme": "",
    "loglevel": 2
}
```
</details>

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox 56.0.2

**Operating system:** Windows 7

#### Nextcloud log (data/nextcloud.log)
```
{"reqId":"l4vlJmddxqBb1XH1nh1E","level":3,"time":"2017-11-09T10:21:29+00:00","remoteAddr":"24.134.13.209","user":"rainer.emrich","app":"index","method":"PUT","url":"\/index.php\/apps\/deck\/cards\/1","message":"Exception: {\"Exception\":\"Doctrine\\\\DBAL\\\\Exception\\\\DriverException\",\"Message\":\"An exception occurred while executing 'UPDATE `oc_deck_cards` SET `duedate` = ?,`last_modified` = ? WHERE `id` = ?' with params [\\\"2017-11-13T10:21:24+00:00\\\", 1510222889, 1]:\\n\\nSQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '2017-11-13T10:21:24+00:00' for column 'duedate' at row 1\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(128): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Statement.php(177): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'UPDATE `oc_deck...', Array)\\n#2 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/public\\\/AppFramework\\\/Db\\\/Mapper.php(247): Doctrine\\\\DBAL\\\\Statement->execute()\\n#3 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/apps\\\/deck\\\/lib\\\/Db\\\/DeckMapper.php(40): OCP\\\\AppFramework\\\\Db\\\\Mapper->execute('UPDATE `*PREFIX...', Array, NULL, NULL)\\n#4 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/public\\\/AppFramework\\\/Db\\\/Mapper.php(187): OCA\\\\Deck\\\\Db\\\\DeckMapper->execute('UPDATE `*PREFIX...', Array)\\n#5 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/apps\\\/deck\\\/lib\\\/Db\\\/CardMapper.php(58): OCP\\\\AppFramework\\\\Db\\\\Mapper->update(Object(OCA\\\\Deck\\\\Db\\\\Card))\\n#6 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/apps\\\/deck\\\/lib\\\/Service\\\/CardService.php(92): OCA\\\\Deck\\\\Db\\\\CardMapper->update(Object(OCA\\\\Deck\\\\Db\\\\Card))\\n#7 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/apps\\\/deck\\\/lib\\\/Controller\\\/CardController.php(94): OCA\\\\Deck\\\\Service\\\\CardService->update(1, 'Das ist eine Un...', 1, 'plain', 999, 'Das ist ein Tes...', 'rainer.emrich', '2017-11-13T10:2...')\\n#8 [internal function]: OCA\\\\Deck\\\\Controller\\\\CardController->update(1, 'Das ist eine Un...', 1, 'plain', 999, 'Das ist ein Tes...', '2017-11-13T10:2...')\\n#9 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(160): call_user_func_array(Array, Array)\\n#10 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(90): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Deck\\\\Controller\\\\CardController), 'update')\\n#11 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(114): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Deck\\\\Controller\\\\CardController), 'update')\\n#12 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('OCA\\\\\\\\Deck\\\\\\\\Contro...', 'update', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#13 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#14 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Router.php(299): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#15 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/lib\\\/base.php(1004): OC\\\\Route\\\\Router->match('\\\/apps\\\/deck\\\/card...')\\n#16 \\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/index.php(48): OC::handleRequest()\\n#17 {main}\",\"File\":\"\\\/var\\\/www\\\/nc_01\\\/nextcloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":116}","userAgent":"Mozilla\/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko\/20100101 Firefox\/56.0","version":"12.0.3.3"}
```

**Comment by juliushaertl** [comment]
@RainerEmrich Thanks for reporting. It seems I somehow missed an important line when backporting the fix to the 0.2 branch. Can you check if the change in #342 fixes the issue for you?

**Comment by RainerEmrich** [comment]
Yeah, #342 fixes the issue for me!
